server/internal/server/vector_store_files.go

```diff
@@ -0,0 +1,281 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"log"
+	"strings"
+
+	fv1 "github.com/llm-operator/file-manager/api/v1"
+	"github.com/llm-operator/rbac-manager/pkg/auth"
+	v1 "github.com/llm-operator/vector-store-manager/api/v1"
+	"github.com/llm-operator/vector-store-manager/server/internal/store"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gorm.io/gorm"
+)
+
+const (
+	vectorStoreFileObject = "vector_store_file"
+
+	minMaxChunkSizeTokens     = int64(100)
+	maxMaxChunkSizeTokens     = int64(4096)
+	defaultMaxChunkSizeTokens = int64(800)
+	defaultChunkOverlapTokens = int64(400)
+)
+
+// CreateVectorStoreFile adds a new file to the vector store.
+func (s *S) CreateVectorStoreFile(
+	ctx context.Context,
+	req *v1.CreateVectorStoreFileRequest,
+) (*v1.VectorStoreFile, error) {
+	userInfo, err := s.extractUserInfoFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.VectorStoreId == "" {
+		return nil, status.Error(codes.InvalidArgument, "vector store id is required")
+	}
+	if req.FileId == "" {
+		return nil, status.Error(codes.InvalidArgument, "file id is required")
+	}
+
+	if err := s.validateVectorStore(req.VectorStoreId, userInfo.ProjectID); err != nil {
+		return nil, err
+	}
+
+	// Pass the Authorization to the context for downstream gRPC calls.
+	ctx = auth.CarryMetadata(ctx)
+
+	resp, err := s.fileWorkerClient.GetFilePath(ctx, &fv1.GetFilePathRequest{
+		Id: req.FileId,
+	})
+	if err != nil {
+		if status.Code(err) == codes.NotFound {
+			return nil, status.Errorf(codes.NotFound, "file %q not found", req.FileId)
+		}
+		return nil, status.Errorf(codes.Internal, "get file path: %s", err)
+	}
+
+	c, err := s.store.GetCollectionByVectorStoreID(userInfo.ProjectID, req.VectorStoreId)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, status.Errorf(codes.NotFound, "vector store %q not found", req.VectorStoreId)
+		}
+		return nil, status.Errorf(codes.Internal, "get collection: %s", err)
+	}
+
+	maxChunkSizeTokens := defaultMaxChunkSizeTokens
+	chunkOverlapTokens := defaultChunkOverlapTokens
+	if cs := req.ChunkingStrategy; cs != nil {
+		if err := validateChunkingStrategy(cs); err != nil {
+			return nil, err
+		}
+		if cs.Static != nil {
+			maxChunkSizeTokens = cs.Static.MaxChunkSizeTokens
+			chunkOverlapTokens = cs.Static.ChunkOverlapTokens
+		}
+	} else {
+		req.ChunkingStrategy = &v1.ChunkingStrategy{
+			Type: string(store.ChunkingStrategyTypeAuto),
```

Comment:
I think we need to convert to `Static`?

If https://platform.openai.com/docs/api-reference/vector-stores-files/file-object is correct, the returned vector store file always have `static`. Adding a TODO comment.

<img width="789" alt="image" src="https://github.com/llm-operator/vector-store-manager/assets/8353550/06d7c77c-58e4-454c-bf35-57f8b0d0a3a2">